DOC-2044 Third Party pack upload steps for Management Appliance (#7458)
spectrocloud/librarium: _partials/self-hosted/management-appliance/_upload-third-party-pack-enablement.mdx

---
partial_category: self-hosted
partial_name: upload-third-party-pack-enablement
---

<Tabs groupId="registry-type">

<TabItem value="internal-zot-registry" label="Internal Zot Registry">

<Tabs>

<TabItem label="Local UI Method" value="local-ui-method">

> 1. Navigate to the [Artifact Studio](https://artifact-studio.spectrocloud.com/) through a web browser and click the **Build bundle** option.

Reviewer: ```suggestion
1. Navigate to the [Artifact Studio](https://artifact-studio.spectrocloud.com/) through a web browser, and under **Create pack bundle**, select **Build bundle**.
```

I got confused and was looking for the section called "Build bundle" 😅 If you make the change, probably update the sentence wherever it occurs, as I know it's in several locations.

> 2. Select the **{props.iso} Appliance** product on the **Product selection** step and select your current version of the **Version selection** step.

Reviewer: ```suggestion
2. Select the **{props.iso} Appliance** product on the **Product selection** step and select your current version on the **Version selection** step.
```

> 3. On the **Use case** step, select **Add-on only** option.

Reviewer: ```suggestion
3. On the **Use case** step, select the **Add-on only** option.
```

> 4. On the **Configure bundle** step, enter **Palette Third Party** in the **Search packs** field and click **Search**. Alternatively, you can find the pack in the **thirdparty** category.

Reviewer: ```suggestion
4. On the **Configure bundle** step, enter `Palette Third Party` in the **Search packs** field and click **Search**. Alternatively, you can find the pack in the **thirdparty** category.
```

Might be me, but I'm also having trouble finding the **thirdparty** category?

Author: It's definitely there.

   Click the checkbox next to the **Palette Third Party** pack to select it, and click **Next Step**.

> 5. On the **Review and download** step, click the **I'm not a robot** reCAPTCHA checkbox, and then click the **Download bundle** button. The download will then start. Alternatively, you can click the **Copy all URLs** button to copy the download URLs to your clipboard.

Reviewer: ```suggestion
5. On the **Review and download** step, click the **I'm not a robot** reCAPTCHA checkbox, and then click the **Download bundle** button to begin the download. Alternatively, you can click the **Copy all URLs** button to copy the download URLs to your clipboard.
```

>    Wait till the pack has been downloaded to your local machine. The pack is downloaded in `.zst` format alongside a signature file in `sig.bin` format.

Reviewer: ```suggestion
   Wait until the pack has been downloaded to your local machine. The pack is downloaded in `.zst` format alongside a signature file in `sig.bin` format.
```

   integrity of the downloaded files using the provided signature file.

   :::

6. Log in to the Local UI of the leader host of the {props.version} management cluster. By default, Local UI is accessible
   at `https://<node-ip>:5080`. Replace `<node-ip>` with the IP address of the leader host.

7. From the left main menu, click **Content**.

8. Click **Actions** in the top right and select **Upload Content** from the drop-down menu.

9. Click the upload icon to open the file selection dialog and select the Third Party ZST file from your local
   machine. Alternatively, you can drag and drop the file into the upload area.

   The upload process starts automatically once the file is selected. You can monitor the upload progress in
   the **Upload Content** dialog.

   Wait for the **Upload Successful** confirmation message to appear.

10. Log in to the {props.version} system console.

11. From the left main menu, select **Administration**, and then select the **Pack Registries** tab.

12. Select the three-dot menu for the **OCI Pack Registry** and click **Sync**.

</TabItem>

<TabItem label="Palette CLI Method" value="palette-cli-method">

> 1. Navigate to the [Artifact Studio](https://artifact-studio.spectrocloud.com/) through a web browser and click the **Build bundle** option.

Reviewer: ```suggestion
1. Navigate to the [Artifact Studio](https://artifact-studio.spectrocloud.com/) through a web browser, and under **Create pack bundle**, select **Build bundle**.
```

> 2. Select the **{props.iso} Appliance** product on the **Product selection** step and select your current version of the **Version selection** step.

Reviewer: ```suggestion
2. Select the **{props.iso} Appliance** product on the **Product selection** step and select your current version on the **Version selection** step.
```

> 3. On the **Use case** step, select **Add-on only** option.

Reviewer: ```suggestion
3. On the **Use case** step, select the **Add-on only** option.
```

> 4. On the **Configure bundle** step, enter **Palette Third Party** in the **Search packs** field and click **Search**. Alternatively, you can find the pack in the **thirdparty** category.

Reviewer: ```suggestion
4. On the **Configure bundle** step, enter `Palette Third Party` in the **Search packs** field and click **Search**. Alternatively, you can find the pack in the **thirdparty** category.
```

   Click the checkbox next to the **Palette Third Party** pack to select it, and click **Next Step**.

> 5. On the **Review and download** step, click the **I'm not a robot** reCAPTCHA checkbox, and then click the **Download bundle** button. The download will then start. Alternatively, you can click the **Copy all URLs** button to copy the download URLs to your clipboard.

Reviewer: ```suggestion
5. On the **Review and download** step, click the **I'm not a robot** reCAPTCHA checkbox, and then click the **Download bundle** button to begin the download. Alternatively, you can click the **Copy all URLs** button to copy the download URLs to your clipboard.
```

>    Wait till the pack has been downloaded to your local machine. The pack is downloaded in `.zst` format alongside a signature file in `sig.bin` format.

Reviewer: ```suggestion
   Wait until the pack has been downloaded to your local machine. The pack is downloaded in `.zst` format alongside a signature file in `sig.bin` format.
```

   --ca-cert <path-to-ca-cert> \
   --tls-cert <path-to-tls-cert> \
   --tls-key <path-to-tls-key>
   ```

   </TabItem>

   <TabItem label="Skip TLS" value="skip-tls">

   ```shell
   palette content push \
   --registry <management-vip>:30003/spectro-content \
   --file <third-party-zst> \
   --insecure
   ```

   </TabItem>

   </Tabs>

   The following example output is expected when the upload is successful.

   ```shell hideClipboard title="Example Output"
   ...
   INFO[0020] successfully copied all artifacts from local bundle /home/ubuntu/palette-cli/bin/tmp/bundle-extract/palette-thirdparty-bundle-4.7.3 to remote bundle 10.11.12.13:30003/spectro-content/bundle-definition:bundle
   -----------------------------
   Push Summary
   -----------------------------
   local bundle bundle pushed to 10.11.12.13:30003/spectro-content
   ```

8. Log in to the {props.version} system console.

9. From the left main menu, select **Administration**, and then select the **Pack Registries** tab.

10. Select the three-dot menu for the **OCI Pack Registry** and click **Sync**.

</TabItem>

</Tabs>

</TabItem>

<TabItem value="external-registry" label="External Registry">

> 1. Navigate to the [Artifact Studio](https://artifact-studio.spectrocloud.com/) through a web browser and click the **Build bundle** option.

Reviewer: ```suggestion
1. Navigate to the [Artifact Studio](https://artifact-studio.spectrocloud.com/) through a web browser, and under **Create pack bundle**, select **Build bundle**.
```

> 2. Select the **{props.iso} Appliance** product on the **Product selection** step and select your current version of the **Version selection** step.

Reviewer: ```suggestion
2. Select the **{props.iso} Appliance** product on the **Product selection** step and select your current version on the **Version selection** step.
```

> 3. On the **Use case** step, select **Add-on only** option.

Reviewer: ```suggestion
3. On the **Use case** step, select the **Add-on only** option.
```

> 4. On the **Configure bundle** step, enter **Palette Third Party** in the **Search packs** field and click **Search**. Alternatively, you can find the pack in the **thirdparty** category.

Reviewer: ```suggestion
4. On the **Configure bundle** step, enter `Palette Third Party` in the **Search packs** field and click **Search**. Alternatively, you can find the pack in the **thirdparty** category.
```

   Click the checkbox next to the **Palette Third Party** pack to select it, and click **Next Step**.

> 5. On the **Review and download** step, click the **I'm not a robot** reCAPTCHA checkbox, and then click the **Download bundle** button. The download will then start. Alternatively, you can click the **Copy all URLs** button to copy the download URLs to your clipboard.

Reviewer: ```suggestion
5. On the **Review and download** step, click the **I'm not a robot** reCAPTCHA checkbox, and then click the **Download bundle** button to begin the download. Alternatively, you can click the **Copy all URLs** button to copy the download URLs to your clipboard.
```

>    Wait till the pack has been downloaded to your local machine. The pack is downloaded in `.zst` format alongside a signature file in `sig.bin` format.

Reviewer: ```suggestion
   Wait until the pack has been downloaded to your local machine. The pack is downloaded in `.zst` format alongside a signature file in `sig.bin` format.
```

   :::tip

   Refer to the <VersionedLink text="Artifact Studio guide" url="/downloads/artifact-studio"/> for detailed guidance on how to verify the
   integrity of the downloaded files using the provided signature file.

   :::

6. Open a terminal on your local machine and navigate to the directory where the Third Party ZST file is located.

7. Authenticate with your external registry using the command line or a GUI tool of your choice. The following examples
   demonstrate how to authenticate with an OCI registry using the [`oras`](https://oras.land/docs/) CLI tool and the Palette CLI.

   <Tabs>

   <TabItem label="Harbor" value="harbor">

   Use `oras` to log in to your OCI registry. Replace the values below with your environment configuration values.
   If you are using a Harbor registry with a self-signed certificate, add the `--insecure` flag to the
   `oras` command.

   <Tabs>

   <TabItem label="Regular Certificate" value="regular-certificate">

   ```shell

>    oras login <registry-url-or-ip> --username '<user-name>' --password '<password>'

Reviewer: ```suggestion
   oras login <registry-url-or-ip> --username '<username>' --password '<password>'
```
